sql/snapshot.sql

```diff
@@ -0,0 +1,100 @@
+-- public.global_snapshot definition
+
+-- Drop table
+
+-- DROP TABLE global_snapshot;
+
+CREATE TABLE global_snapshot (
+	ordinal int8 NOT NULL,
+	hash text NOT NULL,
+	height int8 NOT NULL,
+	subheight int8 NOT NULL,
+	last_snapshot_hash text NULL,
+	created_at timestamp NULL,
+	updated_at timestamp NULL,
```

Comment:
why we need `updated_at` field? afaik all data here are immutable and once stored, never updated 🤔

Reply:
That's not necessarily true. OpenSearch indexes are edited sometimes in unusual cases, for example. The only overhead for `updated_at` is the trigger when it's actually updated so it doesn't really impact performance significantly. It makes the database a lot more human readable in case of any issues though. 

I prefer to have them on all tables but it's up for debate if there's a good reason to remove those columns.

Reply:
> OpenSearch indexes are edited sometimes in unusual cases, for example.

@AlexBrandes could you elaborate on that? what is an example case?

Reply:
You would know better than I would but haven't you guys had to edit or reindex in certain cases? There have been the sporadic issues with balances (I know some of those were issues between 2 indexes). There was also an older case I remember that had a broken chain that had to be fixed. 

I can think of other cases when we might update a row instead of deleting it and re-inserting also. For example, we might decide that we really need all historical state proofs from all snapshots to be returned through the API. Updating the existing rows would be a better strategy than deleting/re-inserting everything. 

I guess the important questions are:
- Will this field impact performance enough that we should avoid it? imo no
- Will this field be useful enough for working with the DB that it's worth including? imo yes

Reply:
AFAIK all cases were that the days were not indexed. Since we ingest blockchain data that are immutable, we should never update data indexed in the database. (For example, an exchange confirmed a transaction and then you update its balance or parent hash, or whatever). AFAIK block explorer, since it's centralized, works as a source of truth for the whole network. Cluster nodes fetch data from block explorer when performing a rollback. Imo those data should never be modified.

Reply:
yeah, I was wondering the same, it was in the initial draft in notion and seemed reasonable :P I'll take them out.

Reply:
So it would be a rare occurrence that a record would be updated - in general I agree. 

What's the argument against including the column though? Performance?

Reply:
Both arguments make sense. Although adding the `updated_at` column later is not free, if we're careful, we can minimize the impact, so I don't think we need to commit right now :) 
(since updates are rare, maybe we can consider a separate table also)